File: solutions/0053-maximum-subarray/solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum=0,max=nums[0];
        for(int i=0;i<nums.size();i++)
        {
            if(sum<0)
                sum=0;
            sum+=nums[i];
            if(max<sum)
                max=sum;
        }
        return max;
    }
};

File: solutions/0053-maximum-subarray/solution.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        sum1=0
        max1=nums[0];
        for i in range(len(nums)):
            if(sum1<0):
                sum1=0;
            sum1+=nums[i];
            if(max1<sum1):
                max1=sum1;
        return max1;
